src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackageDependenciesAndToolsRule.cs

```diff
@@ -67,12 +75,57 @@ public void Apply(MigrationSettings migrationSettings, MigrationRuleInputs migra
                     migrationRuleInputs.OutputMSBuildProject,
                     targetFramework.FrameworkName, 
                     targetFramework.Dependencies,
-                    migrationRuleInputs.ProjectXproj); 
+                    migrationRuleInputs.ProjectXproj);
+
+                if (defaultProjectTargetFramework.IsDesktop())
+                {
+                    allDependencies.AddRange(targetFramework.Dependencies);
+                }
+            }
+
+            // Automatically add the same desktop references that the build for project.json does.
+            // This ensures if the project.json build succeeded then the migrated msbuild one 
+            // will succeed too. See src\Microsoft.DotNet.ProjectModel\Resolution for more details.
+            if (defaultProjectTargetFramework.IsDesktop())
+            {
+                List<ProjectLibraryDependency> autoAddDesktopDependencies = new List<ProjectLibraryDependency>();
+
+                // MSBuild targets automatically add a reference to mscorlib
+                // AddIfMissing(autoAddDesktopDependencies, allDependencies, "mscorlib");
+                AddIfMissing(autoAddDesktopDependencies, allDependencies, "System");
```

Comment:
Can I inject a reference saying if target framework greater than 3.5? Also if there are existing dependencies do I still want to inject? I didn't think so because at that point they are explicitly doing something. But you're saying I always want to inject System and Microsoft.CSharp with the right target framework constraint?